### ISSUE TITLE
Keep tabs in panels consistent across all windows 

### ISSUE THREAD

**Issue by Qjo1** [mbnuqw/sidebery#659]
I'm not sure if this is intended behavior or not so I classified it as a feature request.

## Description

When opening a new window, all panels are empty. Coming from Simple Tab Groups it would be nice if the tabs sorted in panels would remain the same across all windows (or to have an option for it). Tab storage is already set to session storage but switching to global has no effect either.

See attached picture. Left: sorted tabs. Right: new window.

![grafik](https://user-images.githubusercontent.com/107263952/173098649-3b8d1f7b-fe4b-4374-90d9-4c46ec118841.png)

Preferred behavior: have the right side equal the left side.

I looked through settings already but in case I just missed it my apologies and I'll be happy to be directed to it.

P.S.: great add-on!

**Comment by mbnuqw** [comment]
No, it's not possible in v4. In v5 you can manually save tabs panel to bookmarks in one window and restore it in another.

**Comment by Qjo1** [comment]
Yes I saw that function. I tried v5 before posting and it's fantastic. Solved pretty much all issues I had with v4 and looks great. However, using bookmarks like this is at most a workaround in my opinion. For daily work flow it isn't really suitable because you'd have to do that every time you open or close another window. It's right now the main feature keeping me at Simple Tab Groups. 
Unfortunately, my programming knowledge is very limited, otherwise I'd love to investigate how STG handles this.

**Comment by relink2013** [comment]
I came here to ask this exact same question. I actually thought it was a bug and reinstalled Waterfox twice to try and fix it. 

Is there any idea if this feature is coming? Until then My only good option is TST and STG together. I don't know what kind of voodoo magic STG does, but its the only group manager that hasn't randomly lost all my tabs.

**Comment by Qjo1** [comment]
After paying some closer attention to how STG works it seems to handle this by moving all open tabs to the next window upon closing of the current window and then immediately hiding them. It just happens in an instant and is barely visible. And it generally works across multiple windows because STG's groups are basically a collection of which tabs are to be unhidden when this or that group is active, at least that's the impression I have.

This way I can have Group A open in Window 1 and Group B in Window 2 but I can't have Group A open in both windows at the same time. However, the tabs are stored independently, so when I make changes to Group B in Window 2, then switch to Group C in Window 2, I can reopen Group B in Window 1 and it will be consistent.

I too would like to know if we can expect this functionality at some point, because it's a big one in my eyes.

**Comment by robrecord** [comment]
Switching back to STG from Sidebery 5 because I expected/need panels to be global. I come from Safari where this feature is very nicely implemented ([tab groups](https://support.apple.com/en-ca/guide/safari/ibrwa2d73908/mac)).

**Comment by Qjo1** [comment]
> Switching back to STG from Sidebery 5 because I expected/need panels to be global. I come from Safari where this feature is very nicely implemented ([tab groups](https://support.apple.com/en-ca/guide/safari/ibrwa2d73908/mac)).

I actually noticed it isn't that big of a deal. I mean, it would be great to have them global but in the end, opening a panel in a new window or moving tabs over before closing the second window is done in more or less the same amount of clicks as in STG by using drag and drop.

You can drag a panel onto the desktop to open its content in a new window and you can drag existing tabs from one window onto the same panel (does not have to be the active one) in another window. Works for me until we maybe get global tabs in the future.

**Comment by robrecord** [comment]
I don't use drag and drop; My use case is I would open a new window and choose a group (panel), or sometimes I would choose a group (panel) to open in a new window. The first isn't possible AFAICS, the second I'd have to drag out my panel every time.. seems messy.

I have windows separated for each of my client projects, and windows for each major concern (day job, freelance job, one for each web-app I use), and windows for research subjects. 

Being able to close down a named window full of tabs and open it again later is a huge productivity boost.

To this end, I would love a clean, straightforward implementation of Safari tab groups in Firefox.

Are global panels on the roadmap?

**Comment by koppor** [comment]
(A bit off-topic) @Qjo1 I did not find the configuration to show the enivornments as icons on the left. Here, they are only a smal bar at the right side.

![grafik](https://user-images.githubusercontent.com/1366654/210588155-13b30ceb-81e1-44e1-8009-7f5eb9c7aab5.png)

**Comment by ItMeCorban** [comment]
Would love to see this implemented. I too expected it to be the default behaviour given the panels are the same on all windows but after looking through settings concluded that wasn't the case. This would make sidebery a lot more powerful for those of us who live in multiple browser windows.

**Comment by nesanakin** [comment]
Maybe an easier path to this functionality would be to modify Sidebery to interoperate with the Simple Tab Groups extension, which provides a good implementation of whole-window tab groups that work across all open windows, but doesn't provide sub groups or tree tabs. STG already works with Tree Style Tabs. Sidebery doesn't require native tab hiding and has settings to disable it (which are disabled by default) but still seems to have some issues with STG, and is specifically listed as incompatible by STG. I haven't investigated yet to see if there are any fundamental incompatibilities that would make it impossible.

**Comment by ItMeCorban** [comment]
Integration with STG would be a great workaround if it can be done, of course native would be preferred but at the moment I really want to keep using sidebery, but TST/STG combo is sounding really nice for the global groups, despite the lesser (imo) UX.
I'd love to help make this happen if possible, I don't have much dev experience but happy to help with any testing
